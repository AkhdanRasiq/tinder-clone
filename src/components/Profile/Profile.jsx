import React from 'react'

import { Avatar } from '@mui/material'
import LocationOnIcon from '@mui/icons-material/LocationOn';
import EmailIcon from '@mui/icons-material/Email';
import GitHubIcon from '@mui/icons-material/GitHub';
import LinkedInIcon from '@mui/icons-material/LinkedIn';

function Profile() {
  return (
    <div>
      <Avatar className='sidebarProfileAvatar' alt="Avatar Image" src="https://media-exp1.licdn.com/dms/image/C5103AQHB1GRZq-arKg/profile-displayphoto-shrink_400_400/0/1535513202972?e=1648684800&v=beta&t=NFXC30MwzF0Y3zJUgVJk7d3FMAmAHKCdbnESGrOmpYE" />
      <p className='sidebarProfileName'>AKHDAN RASIQ GUMELAR</p>
      <p className='sidebarProfileAbout'>Experienced Software Engineer with a demonstrated history of working in the computer software industry. Skilled in Mobile Application Development, ReactJS, Python, and JavaScript. Strong engineering professional graduated from SMK Telkom Jakarta.</p>
      <div className='sidebarProfilePersonalInfoContainer'>
        <LocationOnIcon fontSize='large' className='sidebarProfilePersonalInfoIcon' />
        <p className='sidebarProfilePersonalInfoText'>Indonesia, Jakarta</p>
      </div>
      <div className='sidebarProfilePersonalInfoContainer'>
        <EmailIcon fontSize='large' className='sidebarProfilePersonalInfoIcon' />
        <p className='sidebarProfilePersonalInfoText'>akhdan.rasiq@gmail.com</p>
      </div>
      <div className='sidebarProfilePersonalInfoContainer'>
        <GitHubIcon fontSize='large' className='sidebarProfilePersonalInfoIcon' />
        <p className='sidebarProfilePersonalInfoText'>github.com/AkhdanRasiq</p>
      </div>
      <div className='sidebarProfilePersonalInfoContainer'>
        <LinkedInIcon fontSize='large' className='sidebarProfilePersonalInfoIcon' />
        <p className='sidebarProfilePersonalInfoText'>linkedin.com/in/AkhdanRasiq</p>
      </div>
    </div>
  )
}

export default Profile
