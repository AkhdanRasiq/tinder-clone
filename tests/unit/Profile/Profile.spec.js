import React from 'react'
import { render } from "@testing-library/react"

import Profile from "../../../src/components/Profile/Profile.jsx"

describe("Profile.jsx", () => {
  it("Render Profile.jsx Properly", async () => {
    const wrapper = render(
      <Profile />
    )
    await expect(wrapper).toMatchSnapshot()
  })
})
