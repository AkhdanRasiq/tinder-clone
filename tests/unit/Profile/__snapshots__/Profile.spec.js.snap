// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile.jsx Render Profile.jsx Properly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="MuiAvatar-root MuiAvatar-circular sidebarProfileAvatar css-1wlk0hk-MuiAvatar-root"
        >
          <img
            alt="Avatar Image"
            class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
            src="https://media-exp1.licdn.com/dms/image/C5103AQHB1GRZq-arKg/profile-displayphoto-shrink_400_400/0/1535513202972?e=1648684800&v=beta&t=NFXC30MwzF0Y3zJUgVJk7d3FMAmAHKCdbnESGrOmpYE"
          />
        </div>
        <p
          class="sidebarProfileName"
        >
          AKHDAN RASIQ GUMELAR
        </p>
        <p
          class="sidebarProfileAbout"
        >
          Experienced Software Engineer with a demonstrated history of working in the computer software industry. Skilled in Mobile Application Development, ReactJS, Python, and JavaScript. Strong engineering professional graduated from SMK Telkom Jakarta.
        </p>
        <div
          class="sidebarProfilePersonalInfoContainer"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge sidebarProfilePersonalInfoIcon css-tzssek-MuiSvgIcon-root"
            data-testid="LocationOnIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 2C8.13 2 5 5.13 5 9c0 5.25 7 13 7 13s7-7.75 7-13c0-3.87-3.13-7-7-7zm0 9.5c-1.38 0-2.5-1.12-2.5-2.5s1.12-2.5 2.5-2.5 2.5 1.12 2.5 2.5-1.12 2.5-2.5 2.5z"
            />
          </svg>
          <p
            class="sidebarProfilePersonalInfoText"
          >
            Indonesia, Jakarta
          </p>
        </div>
        <div
          class="sidebarProfilePersonalInfoContainer"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge sidebarProfilePersonalInfoIcon css-tzssek-MuiSvgIcon-root"
            data-testid="EmailIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4-8 5-8-5V6l8 5 8-5v2z"
            />
          </svg>
          <p
            class="sidebarProfilePersonalInfoText"
          >
            akhdan.rasiq@gmail.com
          </p>
        </div>
        <div
          class="sidebarProfilePersonalInfoContainer"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge sidebarProfilePersonalInfoIcon css-tzssek-MuiSvgIcon-root"
            data-testid="GitHubIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 1.27a11 11 0 00-3.48 21.46c.55.09.73-.28.73-.55v-1.84c-3.03.64-3.67-1.46-3.67-1.46-.55-1.29-1.28-1.65-1.28-1.65-.92-.65.1-.65.1-.65 1.1 0 1.73 1.1 1.73 1.1.92 1.65 2.57 1.2 3.21.92a2 2 0 01.64-1.47c-2.47-.27-5.04-1.19-5.04-5.5 0-1.1.46-2.1 1.2-2.84a3.76 3.76 0 010-2.93s.91-.28 3.11 1.1c1.8-.49 3.7-.49 5.5 0 2.1-1.38 3.02-1.1 3.02-1.1a3.76 3.76 0 010 2.93c.83.74 1.2 1.74 1.2 2.94 0 4.21-2.57 5.13-5.04 5.4.45.37.82.92.82 2.02v3.03c0 .27.1.64.73.55A11 11 0 0012 1.27"
            />
          </svg>
          <p
            class="sidebarProfilePersonalInfoText"
          >
            github.com/AkhdanRasiq
          </p>
        </div>
        <div
          class="sidebarProfilePersonalInfoContainer"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge sidebarProfilePersonalInfoIcon css-tzssek-MuiSvgIcon-root"
            data-testid="LinkedInIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 3a2 2 0 0 1 2 2v14a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2h14m-.5 15.5v-5.3a3.26 3.26 0 0 0-3.26-3.26c-.85 0-1.84.52-2.32 1.3v-1.11h-2.79v8.37h2.79v-4.93c0-.77.62-1.4 1.39-1.4a1.4 1.4 0 0 1 1.4 1.4v4.93h2.79M6.88 8.56a1.68 1.68 0 0 0 1.68-1.68c0-.93-.75-1.69-1.68-1.69a1.69 1.69 0 0 0-1.69 1.69c0 .93.76 1.68 1.69 1.68m1.39 9.94v-8.37H5.5v8.37h2.77z"
            />
          </svg>
          <p
            class="sidebarProfilePersonalInfoText"
          >
            linkedin.com/in/AkhdanRasiq
          </p>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="MuiAvatar-root MuiAvatar-circular sidebarProfileAvatar css-1wlk0hk-MuiAvatar-root"
      >
        <img
          alt="Avatar Image"
          class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
          src="https://media-exp1.licdn.com/dms/image/C5103AQHB1GRZq-arKg/profile-displayphoto-shrink_400_400/0/1535513202972?e=1648684800&v=beta&t=NFXC30MwzF0Y3zJUgVJk7d3FMAmAHKCdbnESGrOmpYE"
        />
      </div>
      <p
        class="sidebarProfileName"
      >
        AKHDAN RASIQ GUMELAR
      </p>
      <p
        class="sidebarProfileAbout"
      >
        Experienced Software Engineer with a demonstrated history of working in the computer software industry. Skilled in Mobile Application Development, ReactJS, Python, and JavaScript. Strong engineering professional graduated from SMK Telkom Jakarta.
      </p>
      <div
        class="sidebarProfilePersonalInfoContainer"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge sidebarProfilePersonalInfoIcon css-tzssek-MuiSvgIcon-root"
          data-testid="LocationOnIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 2C8.13 2 5 5.13 5 9c0 5.25 7 13 7 13s7-7.75 7-13c0-3.87-3.13-7-7-7zm0 9.5c-1.38 0-2.5-1.12-2.5-2.5s1.12-2.5 2.5-2.5 2.5 1.12 2.5 2.5-1.12 2.5-2.5 2.5z"
          />
        </svg>
        <p
          class="sidebarProfilePersonalInfoText"
        >
          Indonesia, Jakarta
        </p>
      </div>
      <div
        class="sidebarProfilePersonalInfoContainer"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge sidebarProfilePersonalInfoIcon css-tzssek-MuiSvgIcon-root"
          data-testid="EmailIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4-8 5-8-5V6l8 5 8-5v2z"
          />
        </svg>
        <p
          class="sidebarProfilePersonalInfoText"
        >
          akhdan.rasiq@gmail.com
        </p>
      </div>
      <div
        class="sidebarProfilePersonalInfoContainer"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge sidebarProfilePersonalInfoIcon css-tzssek-MuiSvgIcon-root"
          data-testid="GitHubIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 1.27a11 11 0 00-3.48 21.46c.55.09.73-.28.73-.55v-1.84c-3.03.64-3.67-1.46-3.67-1.46-.55-1.29-1.28-1.65-1.28-1.65-.92-.65.1-.65.1-.65 1.1 0 1.73 1.1 1.73 1.1.92 1.65 2.57 1.2 3.21.92a2 2 0 01.64-1.47c-2.47-.27-5.04-1.19-5.04-5.5 0-1.1.46-2.1 1.2-2.84a3.76 3.76 0 010-2.93s.91-.28 3.11 1.1c1.8-.49 3.7-.49 5.5 0 2.1-1.38 3.02-1.1 3.02-1.1a3.76 3.76 0 010 2.93c.83.74 1.2 1.74 1.2 2.94 0 4.21-2.57 5.13-5.04 5.4.45.37.82.92.82 2.02v3.03c0 .27.1.64.73.55A11 11 0 0012 1.27"
          />
        </svg>
        <p
          class="sidebarProfilePersonalInfoText"
        >
          github.com/AkhdanRasiq
        </p>
      </div>
      <div
        class="sidebarProfilePersonalInfoContainer"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge sidebarProfilePersonalInfoIcon css-tzssek-MuiSvgIcon-root"
          data-testid="LinkedInIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 3a2 2 0 0 1 2 2v14a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2h14m-.5 15.5v-5.3a3.26 3.26 0 0 0-3.26-3.26c-.85 0-1.84.52-2.32 1.3v-1.11h-2.79v8.37h2.79v-4.93c0-.77.62-1.4 1.39-1.4a1.4 1.4 0 0 1 1.4 1.4v4.93h2.79M6.88 8.56a1.68 1.68 0 0 0 1.68-1.68c0-.93-.75-1.69-1.68-1.69a1.69 1.69 0 0 0-1.69 1.69c0 .93.76 1.68 1.69 1.68m1.39 9.94v-8.37H5.5v8.37h2.77z"
          />
        </svg>
        <p
          class="sidebarProfilePersonalInfoText"
        >
          linkedin.com/in/AkhdanRasiq
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
